webpack/assets/javascripts/react_app/components/componentRegistry.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PieChart from './common/charts/PieChart/';
+import StatisticsChartsList from './statistics/StatisticsChartsList';
+import PowerStatus from './hosts/powerStatus/';
+import NotificationContainer from './notifications/';
+import ToastsList from './toastNotifications/';
+import StorageContainer from './hosts/storage/vmware/';
+import _ from 'lodash';
+
+const componentRegistry = {
+  registry: {},
+
+  register({ name = null, type = null, store = true, data = true }) {
+    if (!name || !type) {
+      throw new Error('Component name or type is missing');
+    }
+    if (this.registry[name]) {
+      throw new Error(`Component name already taken: ${name}`);
+    }
+
+    this.registry[name] = { type, store, data };
+    return this.registry;
+  },
+
+  registerMultiple(componentObjs) {
+    return _.forEach(componentObjs, (obj) => {
```

Comment:
maybe just import foreach instead of the entire lodash, e.g.:

```js
import foreach from 'lodash/foreach'

return foreach(componentObjs, (obj) => {)
```